src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -533,16 +533,16 @@ protected BoundExpression BindExpression(ExpressionSyntax node, BindingDiagnosti
         protected BoundExpression BindExpressionAllowArgList(ExpressionSyntax node, BindingDiagnosticBag diagnostics)
         {
             BoundExpression expr = BindExpressionInternal(node, diagnostics, invoked: false, indexed: false);
-            VerifyUnchecked(node, diagnostics, expr);
+            CheckContextForPointerTypes(node, diagnostics, expr);
             return expr;
         }
 
-        private void VerifyUnchecked(ExpressionSyntax node, BindingDiagnosticBag diagnostics, BoundExpression expr)
+        private void CheckContextForPointerTypes(ExpressionSyntax node, BindingDiagnosticBag diagnostics, BoundExpression expr)
         {
             if (!expr.HasAnyErrors && !IsInsideNameof)
```

Comment:
>if (!expr.HasAnyErrors && !IsInsideNameof) [](http://example.com/codeflow?start=12&length=42)

Would skipping BoundTypeExpression nodes reduce the noise from redundant diagnostics?

Reply:
Possibly. That seems a good idea but I'm trying to keep this PR as small/focused as possible.